vm/vm.go

```diff
@@ -28,17 +31,74 @@ type Engine struct {
 	// prevAssets maps newAsset's inputs by the hash of their PrevID to
 	// their asset.
 	prevAssets commitment.InputSet
+
+	// validateTimeLocks is a flag that indicates whether the engine should
+	// validate lock times.
+	validateTimeLocks bool
```

Comment:
Are we sure we want this flag to exist? Is there still a reason to ever validate without looking at timelocks?

Reply:
Yeah, I think it's nice for unit tests. We would need to hook up way more to get all existing tests passed without the flag.

Reply:
Okay, I flipped the semantics here. Turned out to not be a big change after all.